cypress/tests/widgets/text-input.spec.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+describe('Node/-RED Dashboard 2.0 - Text Input Widget', () => {
+    beforeEach(() => {
+        cy.deployFixture('dashboard-text-input')
+        cy.visit('/dashboard/page1')
+    })
+
+    // Test case: Renders the Text Input widget correctly
+    it('renders the Text Input widget correctly', () => {
+        cy.get('#nrdb-ui-widget-ab3346b81a7cf742 input[type="text"]').should('exist')
+    })
+
+    // Test case: Displays the tooltip correctly
+    it('displays the tooltip correctly', () => {
+        cy.get('#nrdb-ui-widget-ab3346b81a7cf742 .nrdb-ui-text-field').trigger('mouseover')
+        cy.get('.v-tooltip').should('contain', 'Tooltip Text')
+    })
+
+    // Test case: Emits onClear event correctly
+    it('reset field on onClear event correctly and outputs the correct payload', () => {
+        cy.get('#nrdb-ui-widget-ab3346b81a7cf742 input[type="text"]').clear()
+        cy.get('#nrdb-ui-widget-ab3346b81a7cf742 input[type="text"]').type('New text')
+        cy.clickAndWait(cy.get('#nrdb-ui-widget-ab3346b81a7cf742 .nrdb-ui-text-field .v-field__clearable'))
+        cy.get('#nrdb-ui-widget-ab3346b81a7cf742 input[type="text"]').should('have.value', '')
+        cy.checkOutput('msg.payload', null)
+    })
+})
+
+describe('Node-RED Dashboard 2.0 - Text Input (Dynamic Properties)', () => {
+    beforeEach(() => {
+        cy.deployFixture('dashboard-text-input')
+        cy.visit('/dashboard/page1')
+    })
+
+    it('Set the dynamic properties: set input "label"', () => {
+        cy.get('#nrdb-ui-widget-8b7d7031372e4275').should('exist')
+        cy.clickAndWait(cy.get('button').contains('Dynamic Property: Label'))
+        cy.get('#nrdb-ui-widget-8b7d7031372e4275 .v-input .v-field .v-field__field').find('.v-field-label').should('exist')
+    })
+
+    it('Set the dynamic properties: set input "clearable"', () => {
+        cy.get('#nrdb-ui-widget-8b7d7031372e4275').should('exist')
+        cy.clickAndWait(cy.get('button').contains('Dynamic Property: Clearable'))
+        cy.get('#nrdb-ui-widget-8b7d7031372e4275').type('New text')
+        cy.get('#nrdb-ui-widget-8b7d7031372e4275 .v-input .v-field .v-field__field').find('.v-field-label').should('exist')
```

Comment:
what are you actually testing here? It seems to me that it just checks that the `label` element exists, which isn't a useful check in content of whether the `clearable` function is available?

Reply:
You should instead be checking for `v-field__clearable`

Reply:
This is fixed but as usual tests are randomly failing on GH

Local test result 👇

<img width="765" alt="Screenshot 2024-08-21 at 15 44 13" src="https://github.com/user-attachments/assets/ab83f920-98d9-423e-890f-d507f0143cdc">